tests/lib/CoreShop/Test/Data.php

```diff
@@ -53,6 +53,11 @@ class Data
      */
     public static $product3;
 
+    /**
```

Comment:
please don't use PHPUnit for these kind of tests. Most of them get removed with 3.0 as we primarily test with behat.

Reply:
I am not familiar with Behat. It looks like everything is written in Gherkin. So you will translate all tests to behat in 3.0?
In `ShippingTaxationCartItemsTest.php` i wrote a complex mocking test with prophecy and some simpler tests by using the container from Kernel. Shall i remove everything abount testing i added in this PR or only the complex prophecy test? Or did you mean something else?

Reply:
Ah, after thinking a bit further i understand what you mean by not using phpunit tests.
Have you an idea how to test the service? What about testing "update the cart" ?

Reply:
Actually, all of these tests are already done withi behat: https://github.com/coreshop/CoreShop/tree/master/features

We don't do prophecy, you should be able to create the same with behat. So we actually test the behaviour of the application. If you don't know how to do that, I can help you there.

Reply:
I will look at it. Thanks for the moment.

Reply:
Update the cart is already covered. Testing the service directly is not something you would do with behat. You can, but we currently don't. We are testing actually against the cart. So would say something like:


```gherkin
@cart
Feature: Use a different calculation strategy for shipping tax calculation

  Background:
    Given the site operates on a store in "Austria"
    And the site has a currency "Swiss Franc" with iso "CHF"
    And I am in country "Austria"
    And the site has a tax rate "CH" with "7.7%" rate
    And the site has a tax rule group "CH"
    And the tax rule group has a tax rule for country "Austria" with tax rate "CH"
    And the site has a carrier "Post"
    And the carrier "Post" uses tax-calculation strategy "cart-items"
    And adding a shipping rule named "post"
    And the shipping rule is active
    And the shipping rule has a action price of 10 in currency "CHF"
    And the shipping rule belongs to carrier "Post"
    And the site has a product "Shoe" priced at 10000
    And the product has the tax rule group "CH"

  Scenario: Create a new cart, add a product and shipping should be applied
    And I add the product "Shoe" to my cart
    Then the cart shipping should be "1000" excluding tax
    And the cart shipping should be "1077" including tax
    And the loaded carts shipping tax rate should be "7.7"
```

Note that the values are not right there yet, should just get you a feeling how I would dom it